pkg/objectio/column.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objectio
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"github.com/matrixorigin/matrixone/pkg/fileservice"
+	"github.com/matrixorigin/matrixone/pkg/vm/engine/tae/index"
+)
+
+type ColumnBlock struct {
+	meta  *ColumnMeta
+	block *Block
+}
+
+func NewColumnBlock(idx uint16, block *Block) ColumnObject {
+	meta := &ColumnMeta{
+		idx:         idx,
+		zoneMap:     &index.ZoneMap{},
+		bloomFilter: Extent{},
+	}
+	col := &ColumnBlock{
+		block: block,
+		meta:  meta,
+	}
+	return col
+}
+
```

Comment:
This function, (all public functions, unless it is so obvious from name), need a comment.   
What does it read, what is in the result, and if there is any special instructions/notes for error handling.

Reply:
The function here is the implementation of the ColumnObject interface. For comment, see pkg/objectio/type.go